extensions/mssql/src/objectExplorerNodeProvider/webhdfs.ts

```diff
@@ -0,0 +1,717 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as util from 'util';
+import * as url from 'url';
+import * as fs from 'fs';
+import * as querystring from 'querystring';
+import * as request from 'request';
+import * as BufferStreamReader from 'buffer-stream-reader';
+import * as nls from 'vscode-nls';
+import { IHdfsOptions, IRequestParams } from './fileSources';
+
+const localize = nls.loadMessageBundle();
+
+export class WebHDFS {
+	private _requestParams: IRequestParams;
+	private _opts: IHdfsOptions;
+	private _url: any;
+
+	private static ErrorMessageInvalidDataStructure =
+		localize('webhdfs.invalidDataStructure', 'Invalid Data Structure');
+
+	constructor(opts: IHdfsOptions, requestParams: IRequestParams) {
+		if (!(this instanceof WebHDFS)) {
+			return new WebHDFS(opts, requestParams);
+		}
+
+		let missingProps = ['host', 'port', 'path']
+			.filter(p => !opts.hasOwnProperty(p));
+		if (missingProps && missingProps.length > 0) {
+			throw new Error(
+				util.format(
+					localize('webhdfs.missingProperties',
+						'Unable to create WebHDFS client: missing options: %s'),
+					missingProps.join(', ')
+				)
+			);
+		}
+
+		this._requestParams = requestParams;
+		this._opts = opts;
+		this._url = {
+			protocol: opts.protocol || 'http',
+			hostname: opts.host,
+			port: opts.port || 80,
+			pathname: opts.path
+		};
+	}
+
+	private checkArgUndefined(argName: string, argValue: any): void {
+		if (!argValue) {
+			throw new Error(
+				util.format(
+					localize('webhdfs.undefinedArgument', '\'%s\' is undefined.'),
```

Comment:
As mentioned before this isn't how localization works. It needs to be localized with placeholders for values you will insert later. So it'd be as follows:
```suggestion
					localize('webhdfs.undefinedArgument', '\'${0}\' is undefined.', argName),
```

Reply:
fixed